app/controllers/playlists_controller.rb

```diff
@@ -15,16 +15,23 @@ def create
     playlist = Playlist.new(playlist_params)
     if playlist.save
       current_user.playlists << playlist
-      redirect_to user_playlists_path(current_user.id)
+      redirect_to user_playlist_path(current_user.id, playlist)
     else
       flash.now[:danger] = 'Something went wrong. Try making the playlist again'
       render 'new'
     end
   end
 
+  def update
+    playlist = Playlist.find(params[:playlist][:id])
+    playlist.add_song(params[:playlist][:song_id])
+
+    redirect_back(fallback_location: { action: 'show' })
```

Comment:
redirect_back uses a "referrer" header, which may not always present. You should change this to a `redirect_to`. Never assume your users are as cutting-edge as you are. Also, the parentheses are optional.
context:
http://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back
http://www.rubydoc.info/github/rails/rails/ActionController%2FRedirecting%3Aredirect_back